openmc/deplete/abc.py

```diff
@@ -629,6 +645,33 @@ def __init__(
             else:
                 raise ValueError(f"Invalid timestep unit '{unit}'")
 
+        # validate existing depletion steps are consistent with those passed to operator
+        # if the timesteps retrieved from operator.prev_res.get_times match the first set
+        # of time steps provided to the continue run, and the same is true for the source_rates
+        # retrieved from operator.prev_res.get_source_rates(), then run a depletion simulations
+        # with only the new time steps and source rates provided
+        if continue_timesteps:
+            completed_times = operator.prev_res.get_times(time_units=timestep_units)
+            completed_timesteps = completed_times[1:] - completed_times[:-1] # convert absolute t to dt
+            completed_source_rates = operator.prev_res.get_source_rates()
```

Comment:
Let's move this below the timesteps check to put it closer to where it's used. Also, we don't need it if the timesteps check fails.

Reply:
Sorry, where did you want to move this? I currently put it under this loop to convert everything to seconds
```
        for timestep, unit, rate in zip(times, units, source_rates):
```
so we can just use the `seconds` variable.

The idea was to add this right before
```
        self.timesteps = np.asarray(seconds)
        self.source_rates = np.asarray(source_rates)
```
so that we can set the self parameters for a continue run via simple slicing or ignore this block if `continue_timesteps = False`

Reply:
Ah, sorry that was a bit vague. Can we move the declaration of `completed_source_rates` right above the if statement where it's used.